overlord/snapstate/backend/store_info_cache.go

```diff
@@ -0,0 +1,114 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package backend
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+// StoreInfo is information about a snap that is not in the snap.yaml,
+// not needed in the state, but may be cached to augment the information
+// returned for locally-installed snaps
+type StoreInfo struct {
+	Media snap.MediaInfos `json:"media,omitempty"`
+}
+
+func snapStoreInfoCacheFilename(snapName string) string {
+	return filepath.Join(dirs.SnapStoreInfoCacheDir, snapName) + ".json"
+}
+
+func AttachStoreInfo(info *snap.Info) error {
+	if info.SnapID == "" {
+		return nil
+	}
+	snapName := info.InstanceName()
+	f, err := os.Open(snapStoreInfoCacheFilename(snapName))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+	defer f.Close()
+
+	var storeInfo StoreInfo
+	dec := json.NewDecoder(f)
+	if err := dec.Decode(&storeInfo); err != nil {
+		return fmt.Errorf("cannot decode cached store info for snap %q: %v", snapName, err)
+	}
+	if dec.More() {
+		return fmt.Errorf("cannot decode cached store info for snap %q: spurious content after document body", snapName)
+	}
+
+	info.Media = storeInfo.Media
+
+	return nil
+}
+
+func (Backend) CacheStoreInfo(snapName string, storeInfo *StoreInfo) error {
+	if snapName == "" {
+		return nil
+	}
+	if err := os.Mkdir(dirs.SnapStoreInfoCacheDir, 0755); !os.IsExist(err) {
+		if err := os.MkdirAll(dirs.SnapStoreInfoCacheDir, 0755); err != nil {
+			return fmt.Errorf("cannot create directory for store snap info cache: %v", err)
+		}
+		ioutil.WriteFile(filepath.Join(dirs.SnapStoreInfoCacheDir, "README"), []byte(`
```

Comment:
We could ship this file in the packaging but I don't have a strong opinion about it.